experiment/simulation/js/main.js

```diff
@@ -1 +1,554 @@
+      // moving east
+      // mark the moving north east as red
+      highlight(currx + 1, curry + 1, "red");
+      // move east , increment x
+      currx += 1;
+      // y remains the same
+      // highlight the choice
+      highlight(currx, curry, chosen_color);
+    } else {
+      last_move_direction = "north-east";
+      // we move north east
+      // mark east as red
+      highlight(currx + 1, curry, "red");
+      // move north east
+      currx += 1;
+      curry += 1;
+      // highlight the choice
+      highlight(currx, curry, chosen_color);
+      // update the decision parameter
+      decision_parameter = decision_parameter + 2 * dy - 2 * dx;
+    }
+  } else if (slope > 1) {
+    // move north or north east
+    if (decision_parameter < 0) {
+      last_move_direction = "north";
+      decision_parameter = decision_parameter + 2 * dx;
+      // move north
+      highlight(currx + 1, curry + 1, "red");
+      curry += 1;
+      highlight(currx, curry, chosen_color);
+    } else {
+      // moving north east
+      last_move_direction = "north-east";
+      highlight(currx, curry + 1, "red");
+      curry += 1;
+      currx += 1;
+      highlight(currx, curry, chosen_color);
+      decision_parameter = decision_parameter + 2 * dx - 2 * dy;
+    }
+  } else if (slope >= -1 && slope < 0) {
+    // choice to move east or sout east
+    // dx is positive and dy is negative
+    if (decision_parameter < 0) {
+      // move east
+      last_move_direction = "east";
+      highlight(currx + 1, curry - 1, "red");
+      currx += 1;
+      highlight(currx, curry, chosen_color);
+      decision_parameter = decision_parameter + 2 * Math.abs(dy);
+    } else {
+      decision_parameter = decision_parameter + 2 * Math.abs(dy) - 2 * dx;
+      // move south east
+      last_move_direction = "south-east";
+      highlight(currx + 1, curry, "red");
+      curry -= 1;
+      currx += 1;
+      highlight(currx, curry, chosen_color);
+    }
+  } else if (slope === Number.MIN_SAFE_INTEGER) {
+  } else {
+    if (decision_parameter < 0) {
+      last_move_direction = "south";
+      decision_parameter = decision_parameter + 2 * Math.abs(dx);
+      // move south
+      highlight(currx + 1, curry - 1, "red");
+      curry -= 1;
+      highlight(currx, curry, chosen_color);
+    } else {
+      last_move_direction = "south-east";
+      decision_parameter =
+        decision_parameter + 2 * Math.abs(dx) - 2 * Math.abs(dy);
+      // move south east
+      highlight(currx, curry - 1, "red");
+      currx += 1;
+      curry -= 1;
+      highlight(currx, curry, chosen_color);
+    }
+  }
+}
+submit_button.addEventListener("click", (event) => {
+  event.preventDefault();
+  times_next_called = 0;
+  last_move_direction = "";
+  dp = [];
+  point1 = [];
+  point2 = [];
+  slope = 0;
+  currx = 0;
+  curry = 0;
+  decision_parameter = 0;
+  ctx.beginPath();
+  ctx.clearRect(0, 0, width, height);
+  setParameters();
+  if (valid == 1) drawGrid();
```

Comment:
Please use === for comparison
Please create a block for if using {} even if there's only one statement inside it.
The problem with not having a block is that later someone can add code under this line expecting it to run as part of the if conditional and it may introduce bugs.
If you want to use a one-liner, please use valid && drawGrid() so there's no chance of a confusion.

Reply:
done